### ISSUE TITLE
Remote Config: getUserPropertiesWithCallback never executes completion handler

### ISSUE THREAD

**Issue by siddharth952** [firebase/firebase-ios-sdk#9383]
### Step 1: Describe your environment

  * Xcode version: 13.1
  * Firebase SDK version: 8.12.1
  * Installation method: CocoaPods 
  * Firebase Component: Remote Config
  * Target platform(s): iOS

### Step 2: Describe the problem
       

When calling **remoteConfig.fetchAndActivate** in my AppDelegate, it
calls the method **fetchAndActivateWithCompletionHandler** which calls the method **fetchWithCompletionHandler** and passes the closure. Then at the last it executes this the block of code in RCNFetch.m as given below.

as Analytics is installed, getUserPropertiesWithCallback:completionHandler will be called wherein the closure should be handled but it is never executed. 

I have tried to install the Firebase static framework dependencies using Carthage, Zip File and  CocoaPods. I am also using the pods:
Segment-Firebase(https://cocoapods.org/pods/Segment-Firebase)
Segment-appsflyer-ios(https://cocoapods.org/pods/segment-appsflyer-ios)
Segment-WebEngage(https://cocoapods.org/pods/Segment-WebEngage)

#### Steps to reproduce:
debugged it and found -fetchWithUserPropertiesCompletionHandler: method in RCNFetch.m file (Line 227). 

#### Relevant Code:
**AppDelegate:**

```
DispatchQueue.global(qos: .userInteractive).async {
            let remoteConfig = RemoteConfig.remoteConfig()
            let settings = RemoteConfigSettings()
            settings.minimumFetchInterval = Env.activeCase == .debug ? 0 : 14400
           
            remoteConfig.configSettings = settings
            remoteConfig.setDefaults(fromPlist: "ConfigDefaults")
            
           // This function here:
            remoteConfig.fetchAndActivate { (state, error) in
                DispatchQueue.main.async {
                    // Set app values from remoteConfig here ...
                }
            }
      }
```
**RCNFetch.m:** 
 ```
 id<FIRAnalyticsInterop> analytics = self->_analytics;
  if (analytics == nil) { 
    completionHandler(@{});
  } else {
    [analytics getUserPropertiesWithCallback:completionHandler];
  }
```

**Comment by paulb777** [comment]
@siddharth952 Thanks for the report. A few clarification questions:

- When you step through RCNFetch.m, which path does the ` if (analytics == nil)` test take?
- Are there any duplicate symbol warnings in the console on app startup?
- Please share your `Podfile` and `Podfile.lock`

**Comment by siddharth952** [comment]
Hey @paulb777, analytics has a memory address so its not nil and it executes [analytics getUserPropertiesWithCallback:completionHandler]. As for the duplicate symbol warnings we are not getting such warnings.

Podfile:
```
def shared_pods
  pod 'Firebase/Messaging'
  pod 'Firebase/Crashlytics'
  pod 'Firebase/RemoteConfig'
  pod 'Firebase/DynamicLinks'
  pod 'Analytics'
  pod 'WebEngage'
  pod 'GoogleMaps'
  pod 'GooglePlaces'
  pod 'AppsFlyerFramework'
  pod 'Segment-Firebase'
  pod 'segment-appsflyer-ios'
  pod 'Segment-WebEngage'
end

target 'ProjectZ' do
  use_frameworks!  
  shared_pods
end

target 'ProjectZ Debug' do
  use_frameworks!  
  shared_pods
end

target 'ProjectZServiceExtension' do
  use_frameworks! 
  pod 'WebEngageBannerPush'
end

target 'ProjectZContentExtension' do
  use_frameworks! 
  pod 'WebEngageAppEx/ContentExtension'
end
```
Podfile.lock:
```
PODS:
  - Analytics (4.1.6)
  - AppsFlyerFramework (6.5.2):
    - AppsFlyerFramework/Main (= 6.5.2)
  - AppsFlyerFramework/Main (6.5.2)
  - Firebase (8.12.1):
    - Firebase/Core (= 8.12.1)
  - Firebase/Core (8.12.1):
    - Firebase/CoreOnly
    - FirebaseAnalytics (~> 8.12.0)
  - Firebase/CoreOnly (8.12.1):
    - FirebaseCore (= 8.12.1)
  - Firebase/Crashlytics (8.12.1):
    - Firebase/CoreOnly
    - FirebaseCrashlytics (~> 8.12.0)
  - Firebase/DynamicLinks (8.12.1):
    - Firebase/CoreOnly
    - FirebaseDynamicLinks (~> 8.12.0)
  - Firebase/Messaging (8.12.1):
    - Firebase/CoreOnly
    - FirebaseMessaging (~> 8.12.0)
  - Firebase/RemoteConfig (8.12.1):
    - Firebase/CoreOnly
    - FirebaseRemoteConfig (~> 8.12.0)
  - FirebaseABTesting (8.12.0):
    - FirebaseCore (~> 8.0)
  - FirebaseAnalytics (8.12.0):
    - FirebaseAnalytics/AdIdSupport (= 8.12.0)
    - FirebaseCore (~> 8.0)
    - FirebaseInstallations (~> 8.0)
    - GoogleUtilities/AppDelegateSwizzler (~> 7.7)
    - GoogleUtilities/MethodSwizzler (~> 7.7)
    - GoogleUtilities/Network (~> 7.7)
    - "GoogleUtilities/NSData+zlib (~> 7.7)"
    - nanopb (~> 2.30908.0)
  - FirebaseAnalytics/AdIdSupport (8.12.0):
    - FirebaseCore (~> 8.0)
    - FirebaseInstallations (~> 8.0)
    - GoogleAppMeasurement (= 8.12.0)
    - GoogleUtilities/AppDelegateSwizzler (~> 7.7)
    - GoogleUtilities/MethodSwizzler (~> 7.7)
    - GoogleUtilities/Network (~> 7.7)
    - "GoogleUtilities/NSData+zlib (~> 7.7)"
    - nanopb (~> 2.30908.0)
  - FirebaseCore (8.12.1):
    - FirebaseCoreDiagnostics (~> 8.0)
    - GoogleUtilities/Environment (~> 7.7)
    - GoogleUtilities/Logger (~> 7.7)
  - FirebaseCoreDiagnostics (8.12.0):
    - GoogleDataTransport (~> 9.1)
    - GoogleUtilities/Environment (~> 7.7)
    - GoogleUtilities/Logger (~> 7.7)
    - nanopb (~> 2.30908.0)
  - FirebaseCrashlytics (8.12.0):
    - FirebaseCore (~> 8.0)
    - FirebaseInstallations (~> 8.0)
    - GoogleDataTransport (~> 9.1)
    - GoogleUtilities/Environment (~> 7.7)
    - nanopb (~> 2.30908.0)
    - PromisesObjC (< 3.0, >= 1.2)
  - FirebaseDynamicLinks (8.12.0):
    - FirebaseCore (~> 8.0)
  - FirebaseInstallations (8.12.0):
    - FirebaseCore (~> 8.0)
    - GoogleUtilities/Environment (~> 7.7)
    - GoogleUtilities/UserDefaults (~> 7.7)
    - PromisesObjC (< 3.0, >= 1.2)
  - FirebaseMessaging (8.12.0):
    - FirebaseCore (~> 8.0)
    - FirebaseInstallations (~> 8.0)
    - GoogleDataTransport (~> 9.1)
    - GoogleUtilities/AppDelegateSwizzler (~> 7.7)
    - GoogleUtilities/Environment (~> 7.7)
    - GoogleUtilities/Reachability (~> 7.7)
    - GoogleUtilities/UserDefaults (~> 7.7)
    - nanopb (~> 2.30908.0)
  - FirebaseRemoteConfig (8.12.0):
    - FirebaseABTesting (~> 8.0)
    - FirebaseCore (~> 8.0)
    - FirebaseInstallations (~> 8.0)
    - GoogleUtilities/Environment (~> 7.7)
    - "GoogleUtilities/NSData+zlib (~> 7.7)"
  - GoogleAppMeasurement (8.12.0):
    - GoogleAppMeasurement/AdIdSupport (= 8.12.0)
    - GoogleUtilities/AppDelegateSwizzler (~> 7.7)
    - GoogleUtilities/MethodSwizzler (~> 7.7)
    - GoogleUtilities/Network (~> 7.7)
    - "GoogleUtilities/NSData+zlib (~> 7.7)"
    - nanopb (~> 2.30908.0)
  - GoogleAppMeasurement/AdIdSupport (8.12.0):
    - GoogleAppMeasurement/WithoutAdIdSupport (= 8.12.0)
    - GoogleUtilities/AppDelegateSwizzler (~> 7.7)
    - GoogleUtilities/MethodSwizzler (~> 7.7)
    - GoogleUtilities/Network (~> 7.7)
    - "GoogleUtilities/NSData+zlib (~> 7.7)"
    - nanopb (~> 2.30908.0)
  - GoogleAppMeasurement/WithoutAdIdSupport (8.12.0):
    - GoogleUtilities/AppDelegateSwizzler (~> 7.7)
    - GoogleUtilities/MethodSwizzler (~> 7.7)
    - GoogleUtilities/Network (~> 7.7)
    - "GoogleUtilities/NSData+zlib (~> 7.7)"
    - nanopb (~> 2.30908.0)
  - GoogleDataTransport (9.1.2):
    - GoogleUtilities/Environment (~> 7.2)
    - nanopb (~> 2.30908.0)
    - PromisesObjC (< 3.0, >= 1.2)
  - GoogleMaps (6.0.1):
    - GoogleMaps/Maps (= 6.0.1)
  - GoogleMaps/Base (6.0.1)
  - GoogleMaps/Maps (6.0.1):
    - GoogleMaps/Base
  - GooglePlaces (6.0.0)
  - GoogleUtilities/AppDelegateSwizzler (7.7.0):
    - GoogleUtilities/Environment
    - GoogleUtilities/Logger
    - GoogleUtilities/Network
  - GoogleUtilities/Environment (7.7.0):
    - PromisesObjC (< 3.0, >= 1.2)
  - GoogleUtilities/Logger (7.7.0):
    - GoogleUtilities/Environment
  - GoogleUtilities/MethodSwizzler (7.7.0):
    - GoogleUtilities/Logger
  - GoogleUtilities/Network (7.7.0):
    - GoogleUtilities/Logger
    - "GoogleUtilities/NSData+zlib"
    - GoogleUtilities/Reachability
  - "GoogleUtilities/NSData+zlib (7.7.0)"
  - GoogleUtilities/Reachability (7.7.0):
    - GoogleUtilities/Logger
  - GoogleUtilities/UserDefaults (7.7.0):
    - GoogleUtilities/Logger
  - nanopb (2.30908.0):
    - nanopb/decode (= 2.30908.0)
    - nanopb/encode (= 2.30908.0)
  - nanopb/decode (2.30908.0)
  - nanopb/encode (2.30908.0)
  - PromisesObjC (2.0.0)
  - segment-appsflyer-ios (6.5.2):
    - Analytics
    - segment-appsflyer-ios/Main (= 6.5.2)
  - segment-appsflyer-ios/Main (6.5.2):
    - Analytics
    - AppsFlyerFramework (~> 6.5.2)
  - Segment-Firebase (2.7.9):
    - Analytics
    - Firebase (~> 8.7)
    - Firebase/Core (~> 8.7)
    - FirebaseAnalytics (~> 8.7)
    - Segment-Firebase/Core (= 2.7.9)
  - Segment-Firebase/Core (2.7.9):
    - Analytics
    - Firebase (~> 8.7)
    - Firebase/Core (~> 8.7)
    - FirebaseAnalytics (~> 8.7)
  - Segment-WebEngage (1.2.0):
    - Analytics
    - WebEngage (~> 5.2.4)
  - WebEngage (5.2.8):
    - WebEngage/Xcode11 (= 5.2.8)
  - WebEngage/Xcode11 (5.2.8)
  - WebEngageAppEx/ContentExtension (0.12.0):
    - WebEngageAppEx/CoreApi
  - WebEngageAppEx/CoreApi (0.12.0)
  - WebEngageBannerPush (0.12.0)

DEPENDENCIES:
  - Analytics
  - AppsFlyerFramework
  - Firebase/Crashlytics
  - Firebase/DynamicLinks
  - Firebase/Messaging
  - Firebase/RemoteConfig
  - GoogleMaps
  - GooglePlaces
  - segment-appsflyer-ios
  - Segment-Firebase
  - Segment-WebEngage
  - WebEngage
  - WebEngageAppEx/ContentExtension
  - WebEngageBannerPush

SPEC REPOS:
  trunk:
    - Analytics
    - AppsFlyerFramework
    - Firebase
    - FirebaseABTesting
    - FirebaseAnalytics
    - FirebaseCore
    - FirebaseCoreDiagnostics
    - FirebaseCrashlytics
    - FirebaseDynamicLinks
    - FirebaseInstallations
    - FirebaseMessaging
    - FirebaseRemoteConfig
    - GoogleAppMeasurement
    - GoogleDataTransport
    - GoogleMaps
    - GooglePlaces
    - GoogleUtilities
    - nanopb
    - PromisesObjC
    - segment-appsflyer-ios
    - Segment-Firebase
    - Segment-WebEngage
    - WebEngage
    - WebEngageAppEx
    - WebEngageBannerPush

SPEC CHECKSUMS:
  Analytics: eefe524436f904b8bb3f8c8c3425280e43b34efc
  AppsFlyerFramework: 9f304d91cc80b6579b1206a59220383056b58b47
  Firebase: 580d09e8edafc3073ebf09c03fd42e4d80d35fe9
  FirebaseABTesting: 1b05c92327b3f0a9545cea3bab4d93e82550e29b
  FirebaseAnalytics: bd10d7706ba8d6e01ea2816f8fe9e9b881cb0918
  FirebaseCore: 8138de860a90ca7eec5e324da5788fb0ebf1d93c
  FirebaseCoreDiagnostics: 3b40dfadef5b90433a60ae01f01e90fe87aa76aa
  FirebaseCrashlytics: 7952ffd988006128325510664b99dc8ca5851dff
  FirebaseDynamicLinks: 98a5f0fe542245a89ec55f646d2581298c29219d
  FirebaseInstallations: 25764cf322e77f99449395870a65b2bef88e1545
  FirebaseMessaging: 23db8bf05585e929ada8af0f0968933c25252808
  FirebaseRemoteConfig: f174a7091ac8e69bf6d0de2bae2212edb922a0ab
  GoogleAppMeasurement: ae033c3aad67e68294369373056b4d74cc8ae0d6
  GoogleDataTransport: 629c20a4d363167143f30ea78320d5a7eb8bd940
  GoogleMaps: c213fc2334b7374144cede914da779b99be0ab74
  GooglePlaces: c97d96ae68594acb64035c9ca9f26fbfc1d23af1
  GoogleUtilities: e0913149f6b0625b553d70dae12b49fc62914fd1
  nanopb: a0ba3315591a9ae0a16a309ee504766e90db0c96
  PromisesObjC: 68159ce6952d93e17b2dfe273b8c40907db5ba58
  segment-appsflyer-ios: 78c65acfd78dd17ed347623f01919c44b0226174
  Segment-Firebase: 2b07c895603aec168711922fe42a82984375d980
  Segment-WebEngage: 7ef65b70bb1d127743febeea6dbed5e1de3ab9ae
  WebEngage: b01cac28253f3b08b9609c424067d1f9601b2bd1
  WebEngageAppEx: 34f9f4788acff1e517f10a20182a01dd0a34161b
  WebEngageBannerPush: a167dc02c06c568bd7908868bcb1692ea3905719

PODFILE CHECKSUM: 9af8b83b6eb7ba69e2323a6721d211e434228f5c

COCOAPODS: 1.11.2
```

**Comment by paulb777** [comment]
Hmmm, from the code it looks like there should be a console message if the callback doesn't get called. Do you see anything in the console. Does turning on debug logging show anything?   See https://stackoverflow.com/a/55356756/556617

**Comment by siddharth952** [comment]
I cannot see anything regarding duplicates or anything alarming in the console. But the callback should be called as it was working as intended before updating the Firebase dependencies and the segment frameworks.

**Comment by paulb777** [comment]
Strange. Are you able to reproduce in a [quickstart](https://github.com/firebase/quickstart-ios) or share another reproducible example?

**Comment by siddharth952** [comment]
Hey @paulb777 Okay so the callback is now being called when I removed `$(inherited)` from `Other Linker Flags` but now my custom events are not being logged, I have followed the setup guide and tried the possible solutions to the error `[Firebase/Analytics][I-ACS025018] Event not logged. Call +[FIRApp configure]: Test_Event`

But one of my observation is that ApolloCards Framework is using an older version of FirebaseAnalytics (6.17.0) internally. I cannot remove it nor decouple the dependencies it's using as its a compiled static library.

This there a way to avoid this conflict? 

Thanks!
